### ISSUE TITLE
UDF can only be used in SELECT, and abort when used in WHERE, or in INSERT/UPDATE/DELETE commands

### ISSUE THREAD

**Issue by fruch** [scylladb/scylladb#13746]
while trying to port UDF related test I've run into some base test failing, regardless of auth

## setup
```
CREATE KEYSPACE ks WITH REPLICATION = {'class':'SimpleStrategy', 'replication_factor':1}")
CREATE TABLE ks.t1 (k int PRIMARY KEY, v int)")
CREATE FUNCTION ks.plus_one ( input int ) CALLED ON NULL INPUT RETURNS int LANGUAGE lua AS 'return input + 1'
```

## works
- `SELECT k, v, ks.plus_one(v) FROM ks.t1 WHERE k = 1`

## fails with abort
- `SELECT k, v FROM ks.t1 WHERE k = ks.plus_one(0)`
- `INSERT INTO ks.t1 (k, v) VALUES (1, ks.plus_one(1))`
- `UPDATE ks.t1 SET v = ks.plus_one(2) WHERE k = ks.plus_one(0)`
- `DELETE FROM ks.t1 WHERE k = ks.plus_one(0)`

```
failed on teardown with "AssertionError: Critical errors found: [('node1', ['Aborting on shard 1.'])]
Other errors: [('node1', ['ERROR 2023-04-25 13:18:25,435 [shard 1] cql3_fuctions - User function cannot be executed in this context, at: 0x585581e 0x5855dd0 0x58560b8 0x548da87 0x27918e1 0x2791d6c 0x2735431 0x26ff42a 0x275a8dc 0x27650f5 0x275c92e 0x26db02d 0x2912ea2 0x28df8cc 0x28e0033 0x301bdd2 0x24f7bad 0x24f6139 0x251aa2a 0x24eadd7 0x254e002 0x254df55 0x256382e 0x256318b 0x2562580 0x25617b8 0x546a0b3 0x54be3c4 0x54bf647 0x54e1c31 0x548e81a /jenkins/workspace/ub-PRs-scan_scylla-dtest_PR-3140/scylla/.ccm/scylla-repository/124153d439152f09c716fd2f4abf7f1c286a12b1/libreloc/libc.so.6+0x8b12c /jenkins/workspace/ub-PRs-scan_scylla-dtest_PR-3140/scylla/.ccm/scylla-repository/124153d439152f09c716fd2f4abf7f1c286a12b1/libreloc/libc.so.6+0x10cbbf'])]"
```
https://jenkins.scylladb.com/job/sct-github-PRs-scan/job/scylla-dtest/job/PR-3140/1/testReport/auth_roles_test/TestAuthRoles/test_udf_permissions_in_update_2/

Even if we aren't gonna support some things, like WHERE clause, we should return errors, and not abort on those cases.

Migrated to Jira: [SCYLLADB-440](https://scylladb.atlassian.net/browse/SCYLLADB-440)

[SCYLLADB-440]: https://scylladb.atlassian.net/browse/SCYLLADB-440?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by eliransin** [comment]
Of course, this is a plain bug :slightly_smiling_face: 
@cvybhu I guess you will find the spot of this abort in no time .

**Comment by avikivity** [comment]
This is pretty bad.

Let's fix in two phases:
1. throw instead of abort (this can be backported)
2. use with_thread_is_needed if any of there WHERE clause uses UDF

**Comment by nyh** [comment]
@avikivity note that this message is already an `on_internal_error()`, which is already a throw and not an abort (we decided to be over-zealos in our tests and force aborts, I still think it's a bad idea, but anyway an ordinary user will not see this as an abort by default, and your "phase 1" is not needed.

But of course we shouldn't have reached this `on_internal_error()` and the fact we did is a bug that should be fixed.

**Comment by mykaul** [comment]
@wmitros - it'd be great if we can at least fix the first part here ( https://github.com/scylladb/scylladb/issues/13746#issuecomment-1534894244 )

**Comment by nyh** [comment]
By the way, I think this become an on_internal_error() after https://github.com/scylladb/scylladb/issues/7977 was "fixed" but not really fixed.
Believe it or not, we even have a unit test enshrining this failure,  test/boost/user_function_test.cc, test_user_function_filtering :-( This was a mistake. We should have fixed the bug, not write a test that ensures we didn't fix it by mistake. If this request is an error (is it an error in Cassandra?), it should be a normal error, not an on_internal_error().

**Comment by nyh** [comment]
> @wmitros - it'd be great if we can at least fix the first part here ( [#13746 (comment)](https://github.com/scylladb/scylladb/issues/13746#issuecomment-1534894244) )

"The first part" - throwing instead of aborting - was already done in #7977 when on_internal_error() was used: on_internal_error() throws, it doesn't abort, except in our test framework (as I noted above, I don't like this fact, but it's true). Maybe you mean throw a more "regular" type of error. But what we really should do is check what Cassandra does in this case, and do the same. If we don't support something yet which is supported in Cassandra., let's throw an error explicitly saying it's not supported yet.

**Comment by dani-tweig** [comment]
Closing this issue as it was moved to Jira.
Please continue the thread in https://scylladb.atlassian.net/browse/SCYLLADB-440

**Comment by nyh** [comment]
I'll not reopen this issue because I wasn't the one to create it, but since @ptrsmrn hasn't been working on it I don't understand why he asked to migrate it.

@dani-tweig did you look at what the migrated issues look like? does it not bother you that the links in the migrated issues no longer work like they did in github, and in general the issue doesn't look as good as it did in github?

**Comment by dani-tweig** [comment]
> I'll not reopen this issue because I wasn't the one to create it, but since [@ptrsmrn](https://github.com/ptrsmrn) hasn't been working on it I don't understand why he asked to migrate it.
> 
> [@dani-tweig](https://github.com/dani-tweig) did you look at what the migrated issues look like? does it not bother you that the links in the migrated issues no longer work like they did in github, and in general the issue doesn't look as good as it did in github?

I know that you dont like the migration to Jira, and that the migrated issue isn't as pretty and functional as the original, and this is why in scylladb.git we chose to only cherry-pick migrate issues based on specific requests.

**Comment by nyh** [comment]
> I know that you dont like the migration to Jira, and that the migrated issue isn't as pretty and functional as the original, and this is why in scylladb.git we chose to only cherry-pick migrate issues based on specific requests.

Yes, I realize this was @ptrsmrn's request and I wonder why (I'm only commenting or reopening issues where I was personally involved and @ptrsmrn wasn't really - I'm not out to second-guess every one of his migration requests).

But I'm more bothered by the fact that as you say "the migrated issue isn't as pretty and functional as the original" but we just take it as a fact of life. Force majeure. As if this migration, or its implementation, was forced on us and not our own doing. So we don't care that the migrated issue has links that don't work. We don't care that xfailing tests refer to the old issue, not the new one. We don't care that migrated issues have comments from "IT admin" from "an hour ago" instead of being properly back-dated. Well, I do care, and I can't stop caring just because "it was decided", sorry about that.

**Comment by ptrsmrn** [comment]
This item is part of the CQL Evolution effort, and that's why it's been migrated: https://github.com/scylladb/scylladb/issues/2693#issuecomment-3816108081
There's no known better way to migrate issues to Jira, but we'll keep both Jira and GH issue, so the history etc are not lost.
